src/app/shared/spaces.service.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { TestBed } from '@angular/core/testing';
+
+import { ProfileService, ExtProfile } from './../profile/profile.service';
+import { Broadcaster } from 'ngx-base';
+import { ConnectableObservable } from 'rxjs/observable/ConnectableObservable';
+import { ReplaySubject } from 'rxjs/ReplaySubject';
+import { Injectable } from '@angular/core';
+
+import { LocalStorageService } from 'angular-2-local-storage';
+import { Observable, Subject, BehaviorSubject } from 'rxjs';
+import { Spaces, Contexts, Space, SpaceService } from 'ngx-fabric8-wit';
+import { cloneDeep } from 'lodash';
+
+import { SpacesService } from './spaces.service';
+import {context1, context2, space, profile } from './spaces.service.mock';
+
+describe('Spaces Service:', () => {
+	let mockContexts: any;
+	let mockLocalStorage: any;
+  let mockSpaceService: any;
+  let mockProfileService: any;
+  let spacesService: SpacesService;
+  let broadcaster: Broadcaster;
+
+	beforeEach(() => {
+		mockContexts = jasmine.createSpyObj('Contexts', ['current']);
+		mockContexts.current = Observable.of(context1);
+    mockLocalStorage = jasmine.createSpy('LocalStorageService');
+		mockSpaceService = jasmine.createSpyObj('mockSpaceService', ['getSpaceById']);
+		mockSpaceService.getSpaceById = (spaceId: string): Observable<Space> => {
+			let tmpspace = cloneDeep(space);
+			tmpspace.name = "space_" + spaceId;
+			tmpspace.id = spaceId;
+			tmpspace.attributes.name = "space_" + spaceId;
+			return  Observable.of(tmpspace);
+    };
+
+    mockProfileService = jasmine.createSpy('ProfileService');
+    mockProfileService.current = Observable.of(profile);
+
+    TestBed.configureTestingModule({
```

Comment:
How about using angular component to test service with it together? I assume this service is used by some component so we could test both the service and ui component behaviour. For this purpose you can have a look at `TestContext` which simplifies usage of `TestBed` https://github.com/fabric8-ui/fabric8-ui/commit/04d18c56ab9ef766637ef34253ad4b1a4efea6ca#diff-6f9e140c353109a019c0342cc6d44aa6R25